### ISSUE TITLE
Add "no guests" config setting

### ISSUE THREAD

**Issue by 1eob** [BeamMP/BeamMP-Server#247]
It would be nice to see a setting within the server configuration which disallows guest users from joining. 

The setting name would be like `authentication_required = true | false`

Ideally, this setting would also be communicated through heartbeat, so that end users would know that they would be able to join the server or not through the server list.

Acceptance Criteria
---
- [ ] Add "no guests" config setting
- [ ] Add equivalent ENV setting
- [ ] Send via heartbeat

**Comment by lionkor** [comment]
Related to https://github.com/BeamMP/BeamMP/issues/469 and https://github.com/BeamMP/Backend/issues/33

**Comment by lionkor** [comment]
@jimkoen please be aware that the implementation of denying guests, which happens in TNetwork::Authentication, will have to be reworked for 4.0.0 -- Of course I'm happy if this feature makes in in first, and most of the code is adding the config settings. We'll need to make a new issue once this is done to track also implementing that on the new protoocol
